src/db.rs

```diff
@@ -2270,6 +2279,35 @@ fn iterator_test() {
     assert!(DB::destroy(&opts, path).is_ok());
 }
 
+#[test]
+fn iterator_test_tailing() {
+    let path = "_rust_rocksdb_iteratortest_tailing";
+    {
+        let mut ro = ReadOptions::default();
+        ro.set_tailing(true);
+        let db = DB::open_default(path).unwrap();
+        let p = db.put(b"k1", b"v1111");
+        assert!(p.is_ok());
+        let iter = db.iterator_opt(IteratorMode::Start, &ro);
+        let p = db.put(b"k2", b"v2222");
+        assert!(p.is_ok());
+        let p = db.put(b"k3", b"v3333");
+        assert!(p.is_ok());
+        let mut i = 0;
+        for (k, v) in iter {
+            println!(
+                "Hello {}: {}",
+                str::from_utf8(&*k).unwrap(),
+                str::from_utf8(&*v).unwrap()
+            );
+            i = i + 1;
+        }
+        assert_eq!(i, 3);
+    }
+    let opts = Options::default();
+    assert!(DB::destroy(&opts, path).is_ok());
+}
```

Comment:
I'd prefer to see something like this instead: 
```rust
#[test]
fn iterator_test_tailing() {
    let path = "_rust_rocksdb_iteratortest_tailing";
    {
        let data = [(b"k1", b"v1"), (b"k2", b"v2"), (b"k3", b"v3")];
        let mut ro = ReadOptions::default();
        ro.set_tailing(true);
        let db = DB::open_default(path).unwrap();

        let (key, value) = data[0];
        let p = db.put(key, value);
        assert!(p.is_ok());

        let iter = db.iterator_opt(IteratorMode::Start, &ro);

        let (key, value) = data[1];
        let p = db.put(key, value);
        assert!(p.is_ok());

        let (key, value) = data[2];
        let p = db.put(key, value);
        assert!(p.is_ok());

        for (i, (k, v)) in iter.enumerate() {
            let (real_key, real_value) = data[i];
            assert_eq!(
                (real_key.to_vec(), real_value.to_vec()),
                (k.to_vec(), v.to_vec())
            );
        }
    }
    let opts = Options::default();
    assert!(DB::destroy(&opts, path).is_ok());
}
```

What do you thinks ?

Reply:
Sure, I can definitely do something like that. I had something similar at first, but then abandoned it to be like the iterator test above it. I'll update the PR this evening, PST time.